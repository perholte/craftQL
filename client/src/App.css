html,
body {
    margin: 0px;
}

body,
#root {
    background-color: rgba(232, 205, 102, 255);
}
