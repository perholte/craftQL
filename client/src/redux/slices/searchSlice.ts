import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { FilterField } from '../../generated/graphql';

export interface SearchState {
    value: string;
    field: FilterField;
}

const initialState: SearchState = {
    value: '',
    field: FilterField.Name,
};

export const counterSlice = createSlice({
    name: 'search',
    initialState,
    reducers: {
        setSearchValue: (state, action: PayloadAction<string>) => {
            state.value = action.payload;
        },
        setSearchCategory: (state, action: PayloadAction<FilterField>) => {
            state.field = action.payload;
        },
        clearSearch: (state) => {
            state.value = '';
            state.field = FilterField.Name;
        },
    },
});

// Action creators are generated for each case reducer function
export const { setSearchValue, setSearchCategory } = counterSlice.actions;

export default counterSlice.reducer;
