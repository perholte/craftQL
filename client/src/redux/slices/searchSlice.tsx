import { createSlice, PayloadAction } from '@reduxjs/toolkit';

export interface SearchState {
    value: string;
}

const initialState: SearchState = {
    value: '',
};

export const counterSlice = createSlice({
    name: 'search',
    initialState,
    reducers: {
        setSearchValue: (state, action: PayloadAction<string>) => {
            state.value = action.payload;
        },
        clearSearch: (state) => {
            state.value = '';
        },
    },
});

// Action creators are generated for each case reducer function
export const { setSearchValue } = counterSlice.actions;

export default counterSlice.reducer;
