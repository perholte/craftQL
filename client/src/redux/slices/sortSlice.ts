import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { BeerOrder, Sort } from '../../generated/graphql';

export enum SortOptions {
    AlphabeticASC = 'Name: A - Z',
    AlphabeticDESC = 'Name: Z - A',
    BrandASC = 'Brand: A - Z',
    BrandDESC = 'Brand: Z - A',
    TypeASC = 'Type A - Z',
    TypeDESC = 'Type Z - A',
    AlcoholASC = 'Most alcohol',
    AlcoholDESC = 'Least alcohol',
    RatingDESC = 'Highest rating',
    RatingASC = 'Lowest rating',
}

const sortOptionToGraphqlParams = new Map<SortOptions, BeerOrder>([
    [SortOptions.AlphabeticASC, { name: Sort.Asc }],
    [SortOptions.AlphabeticDESC, { name: Sort.Desc }],
    [SortOptions.BrandASC, { brand: Sort.Asc }],
    [SortOptions.BrandDESC, { brand: Sort.Desc }],
    [SortOptions.TypeASC, { type: Sort.Asc }],
    [SortOptions.TypeDESC, { type: Sort.Desc }],
    [SortOptions.AlcoholASC, { abv: Sort.Asc }],
    [SortOptions.AlcoholDESC, { abv: Sort.Desc }],
    [SortOptions.RatingASC, { rating: Sort.Asc }],
    [SortOptions.RatingDESC, { rating: Sort.Desc }],
]);

export interface SortState {
    sortOption: SortOptions;
    graphqlParams: BeerOrder;
}

const initialState: SortState = {
    sortOption: SortOptions.AlphabeticASC,
    graphqlParams: sortOptionToGraphqlParams.get(SortOptions.AlphabeticASC) as BeerOrder,
};

export const sortSlice = createSlice({
    name: 'sort',
    initialState,
    reducers: {
        setSortingChoice: (state, action: PayloadAction<SortOptions>) => {
            state.sortOption = action.payload;
            state.graphqlParams = sortOptionToGraphqlParams.get(action.payload) as BeerOrder;
        },
    },
});

// Action creators are generated for each case reducer function
export const { setSortingChoice } = sortSlice.actions;

export default sortSlice.reducer;
