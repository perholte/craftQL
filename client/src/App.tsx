import React from 'react';
import './App.css';
import Header from './components/header/Header';
import Homepage from './components/homepage/Homepage';
import SearchPage from './components/searchPage/SearchPage';

function App() {
    return (
        <div className="app">
            <Header />
            <SearchPage />
        </div>
    );
}

export default App;
