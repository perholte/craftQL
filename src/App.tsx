import React from 'react';
import './App.css';
import Header from './components/header/Header';
import Homepage from './components/homepage/Homepage';

function App() {
    return (
        <div className="app">
            <Header />
            <Homepage />
        </div>
    );
}

export default App;
