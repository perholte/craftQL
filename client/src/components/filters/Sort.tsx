import { FormControl, FormLabel, Select } from '@chakra-ui/react';
import React, { ChangeEvent } from 'react';
import { useSelector } from 'react-redux';
import { setSortingChoice, SortOptions } from '../../redux/slices/sortSlice';
import { RootState, useAppDispatch } from '../../redux/store';

const Sort: React.FC = () => {
    const dispatch = useAppDispatch();
    const selected = useSelector((state: RootState) => state.sort.sortOption);

    const handleChange = (e: ChangeEvent<HTMLSelectElement>) => {
        dispatch(setSortingChoice(e.target.value as SortOptions));
    };

    return (
        <FormControl w="auto" minW="100px" pt="1rem">
            <FormLabel>Sort beers by: </FormLabel>
            <Select onChange={handleChange} value={selected}>
                {Object.values(SortOptions).map((so) => (
                    <option value={so} key={so}>
                        {so}
                    </option>
                ))}
            </Select>
        </FormControl>
    );
};

export default Sort;
