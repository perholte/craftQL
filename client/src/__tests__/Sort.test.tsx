import { mount, shallow } from 'enzyme';
import { Provider } from 'react-redux';
import Sort from '../components/filters/Sort';
import { store } from './App.test';
import { Select } from '@chakra-ui/react';
import { SortOptions } from '../redux/slices/sortSlice';

describe('<Sort />', () => {
    const wrapper = mount(
        <Provider store={store}>
            <Sort />
        </Provider>,
    );

    it('renders successfully', () => {
        expect(wrapper).toHaveLength(1);
    });

    it('can change sortable value', () => {
        wrapper.find('select').simulate('change', { target: { value: 'Name: Z - A' } });
        expect(wrapper.find(Select).props().value).toEqual('Name: Z - A');
        wrapper.find('select').simulate('change', { target: { value: 'Most alcohol' } });
        expect(wrapper.find(Select).props().value).toEqual('Most alcohol');
    });
});
/*     AlphabeticASC = 'Name: A - Z',
    AlphabeticDESC = 'Name: Z - A',
    BrandASC = 'Brand: A - Z',
    BrandDESC = 'Brand: Z - A',
    TypeASC = 'Type A - Z',
    TypeDESC = 'Type Z - A',
    AlcoholASC = 'Most alcohol',
    AlcoholDESC = 'Least alcohol',

    });
 */
