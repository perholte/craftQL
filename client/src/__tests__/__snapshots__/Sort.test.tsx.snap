// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders app 1`] = `ShallowWrapper {}`;
